components/skus/browser/skus_sdk_fetcher_impl.cc

```diff
@@ -0,0 +1,111 @@
+
+SkusSdkFetcherImpl::SkusSdkFetcherImpl(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : url_loader_factory_(url_loader_factory) {}
+
+SkusSdkFetcherImpl::~SkusSdkFetcherImpl() {}
+
+void SkusSdkFetcherImpl::BeginFetch(
+    const skus::HttpRequest& req,
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<skus::HttpRoundtripContext>,
+             skus::HttpResponse)> callback,
+    rust::cxxbridge1::Box<skus::HttpRoundtripContext> ctx) {
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = GURL(static_cast<std::string>(req.url));
+  resource_request->method = static_cast<std::string>(req.method).c_str();
+  resource_request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  // No cache read, always download from the network.
+  resource_request->load_flags =
+      net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE;
+
+  for (size_t i = 0; i < req.headers.size(); i++) {
+    resource_request->headers.AddHeaderFromString(
+        static_cast<std::string>(req.headers[i]));
+  }
+
+  sku_sdk_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), GetNetworkTrafficAnnotationTag());
+
+  sku_sdk_loader_->DownloadToString(
+      url_loader_factory_.get(),
+      base::BindOnce(&SkusSdkFetcherImpl::OnFetchComplete,
+                     base::Unretained(this), std::move(callback),
+                     std::move(ctx)),
+      kMaxResponseSize);
+}
+
+const net::NetworkTrafficAnnotationTag&
+SkusSdkFetcherImpl::GetNetworkTrafficAnnotationTag() {
+  static const net::NetworkTrafficAnnotationTag network_traffic_annotation_tag =
+      net::DefineNetworkTrafficAnnotation("sku_sdk_execute_request", R"(
+      semantics {
+        sender: "Brave SKU SDK"
+        description:
+          "Call the SKU SDK implementation provided by the caller"
+        trigger:
+          "Any Brave webpage using SKU SDK where window.brave.sku.*"
+          "methods are called; ex: fetch_order / fetch_order_credentials"
+        data: "JSON data comprising an order."
+        destination: OTHER
+        destination_other: "Brave developers"
+      }
+      policy {
+        cookies_allowed: NO
+      })");
+  return network_traffic_annotation_tag;
+}
+
+void SkusSdkFetcherImpl::OnFetchComplete(
+    rust::cxxbridge1::Fn<
+        void(rust::cxxbridge1::Box<skus::HttpRoundtripContext>,
+             skus::HttpResponse)> callback,
+    rust::cxxbridge1::Box<skus::HttpRoundtripContext> ctx,
+    std::unique_ptr<std::string> response_body) {
+  if (!response_body) {
+    std::vector<uint8_t> body_bytes;
+    skus::HttpResponse resp = {
+        skus::SkusResult::RequestFailed,
+        500,
+        {},
+        body_bytes,
+    };
+    callback(std::move(ctx), resp);
+    return;
+  }
+
+  std::vector<uint8_t> body_bytes(response_body->begin(), response_body->end());
+
+  skus::HttpResponse resp = {
+      skus::SkusResult::Ok,
+      200,
```

Comment:
same as above, we should be returning the actual error code if nothing else to validate the response

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/c5093d6de8ced06ed9eeae17bfc7545e3ce0dc89